### DISCUSSION TITLE
Document related with network

### DISCUSSION THREAD

**Discussion by SamYuan1990** [containers/podman#13441] [Ideas]
Hello everyone,

I took a look at with network related things today.

- We organize document as 
```
network command overall
network subcommand as create, list ...
```

- We have content for linux, mac, windows and remote client, in 
```
remote client
mac guidance...
```

- We have different CNI provider.
```
Netavark
CNI
MacVLAN
```

As far as I learned from code, in podman, we followed client server mode, and the key line to interact from client to server for network related thing is:
```
	response, err := registry.ContainerEngine().NetworkCreate(registry.Context(), network)
```

Which means:
1.  For linux, mac, windows and remote client, it's able to be covered in `registry.ContainerEngine()` related document.
2. For CNI providers it's able to be covered in `NetworkCreate(registry.Context(), network)`

I am not sure if we are welcome/free to 

- [ ] add a new part as point 1 above to have a specific document discussion with how to configure a remote sever for podman, with knowledges.
- General steps to connect with any remote client.
- Take linux, mac, windows as sample, discussion specific and different default approach. For ex mac with fedora image.

- [ ] In Network general part, adding a general description part with `registry.ContainerEngine().` approach and moving `Netavark, CNI, MacVLAN` from `network-create` to `network` as overall/high level design. 
- [ ] In Create network part/a new page/network general description add some comparisons between different "CNI" as `Netavark, CNI, MacVLAN` in a general speaking.

**Comment by mheon** [comment]
@TomSweeneyRedHat PTAL

**Comment by Luap99** [comment]
What kind of documentation do you want? Developer or end user?

Right now you are mixing terms here. With podman 4 we support 2 network backends (`CNI` and `netavark`). Both backends support drivers (`bridge` and `macvlan`).
You find the podman4 network code here https://github.com/containers/common/tree/main/libnetwork
CNI Repos: https://github.com/containernetworking/plugins https://github.com/containernetworking/cni
Netavark: https://github.com/containers/netavark and aardvark-dns: https://github.com/containers/aardvark-dns/

**Reply by SamYuan1990** [reply]
well, 1st I suppose developer is one of end user, as developer who use docker/container technologic.


in my point of view, as a developer, when he/she decide to chose one from table below as container network.
| network provider | network type | 
| ---- | ---- | 
| CNI | macvlan |
| CNI | bridge |
| netavark | macvlan |
| netavark | bridge  |

He/She is better to know as much technological detail as possible to support the decision making.

I am agree with that we should figure out or point out what's podman document and what's not. 
but... as document as:
https://github.com/containers/netavark/blob/main/docs/netavark.1.md
seems not have too much details with different `macvlan` mode and `bridge` mode for `netavark`
